Lib/tracemalloc.py

```diff
@@ -182,15 +182,20 @@ class Traceback(Sequence):
     Sequence of Frame instances sorted from the oldest frame
     to the most recent frame.
     """
-    __slots__ = ("_frames",)
+    __slots__ = ("_frames", '_total_nframe')
 
-    def __init__(self, frames):
+    def __init__(self, frames, total_nframe):
```

Comment:
What do you think of making this new parameter optional (total_nframe=None), and use len(frames) by default, if it's not specified? It would simplify get_object_traceback() and test_tracemalloc.

Reply:
That means you lose information when you load traces from Python <= 3.8 as there's no way to know if the `total_nframe` was captured or simply faked at load time.
I think it's better to say that you either that information or you don't (so set to `None`) rather than trying to make the field ambiguous.